modules/publicize/publicize-jetpack.php

```diff
@@ -156,11 +156,12 @@ function admin_page_load() {
 				$wpcom_blog_id = !empty( $wpcom_blog_id ) ? $wpcom_blog_id : $stats_options['blog_id'];
 
 				$user = wp_get_current_user();
+				$custom_site_url = apply_filters( 'publicize_custom_site_url', site_url() );
 				$redirect = $this->api_url( $service_name, urlencode_deep( array(
 					'action'       => 'request',
 					'redirect_uri' => add_query_arg( array( 'action' => 'done' ), menu_page_url( 'sharing', false ) ),
 					'for'          => 'publicize', // required flag that says this connection is intended for publicize
-					'siteurl'      => site_url(),
+					'siteurl'      => $custom_site_url,
```

Comment:
I think this should just use `home_url()` instead of `site_url()`. The filter seems a bit sketchy to me.